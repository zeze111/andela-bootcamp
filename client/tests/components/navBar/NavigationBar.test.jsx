import React from 'react';

import connectedNavigationBar, {NavigationBar} from '../../../components/navbar/NavigationBar';
import store from '../../../store';

const props = {
  signout: jest.fn(() => Promise.resolve()),
  auth: {},
  user: {}
};

const context = {
  router: {
    history: {
      push: jest.fn()
    }
  }
};

const setup = () => {
  const wrapper = shallow(<NavigationBar {...props} store={store} />, {context});
  return { wrapper }
}

describe('component will receive props', () => {
  it('should call get recipes function', () => {
    const spy = sinon.spy(NavigationBar.prototype, 'componentWillReceiveProps');

    const nextProps = {
      signout: jest.fn(() => Promise.resolve()),
      auth: {},
      user: {
        firstName: 'Jane'
      }
    };

    const wrapper = shallow(<NavigationBar {...nextProps} />);
    wrapper.setProps({ prop: 3 });
    expect(spy.calledOnce).toEqual(true);
  });
});

describe('signout', () => {
  it('be called when form is submitted', () => {
    const { wrapper } = setup();

    wrapper.instance().signout();
  });
});

describe('Test for Navigation Bar', () => {
  it('should render component', () => {
    const wrapper = setup();
    document.body.innerHTML =
    '<div>' +
    '  <Link class="button-collapse" />' +
    '  <div class="tooltipped" />' +
    '</div>';

    expect(wrapper).toMatchSnapshot()
  });
});
