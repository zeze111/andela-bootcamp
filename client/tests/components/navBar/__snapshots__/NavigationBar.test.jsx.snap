// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Navigation Bar should render component 1`] = `
Object {
  "wrapper": <header>
    <nav
      className="nav-color"
    >
      <div
        className="nav-wrapper "
      >
        <Link
          className="left brand-logo logo-style"
          href="/"
          id="logo"
          replace={false}
          to="/"
        >
          MoreRecipes
        </Link>
        <Link
          className="button-collapse left"
          data-activates="nav-mobile"
          href="#!"
          replace={false}
          to="#!"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </Link>
        <div>
          <ul
            className="right hide-on-med-and-down nav-style"
          >
            <li>
              <Link
                href="/signup"
                replace={false}
                to="/signup"
              >
                Sign Up / In
              </Link>
            </li>
          </ul>
          <ul
            className="side-nav"
            id="nav-mobile"
          >
            <li
              className="nav-list"
            >
              <i
                className="material-icons"
              >
                 keyboard_tab 
              </i>
              <Link
                className="white-text list-item2"
                href="/signup"
                replace={false}
                to="/signup"
              >
                Sign Up / In
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>,
}
`;
